lib/floe/workflow/intrinsic_function/transformer.rb

```diff
@@ -0,0 +1,31 @@
+# Disable rubocops against the `match` method, since this is a Parslet specific
+#   match method and not the typical `Object#match`.
+# rubocop:disable Performance/RegexpMatch, Performance/RedundantMatch
+
+require "parslet"
+require "jsonpath"
+require "securerandom"
+
+module Floe
+  class Workflow
+    module IntrinsicFunction
+      class Transformer < Parslet::Transform
+        rule(:null_literal  => simple(:v)) { nil }
+        rule(:true_literal  => simple(:v)) { true }
+        rule(:false_literal => simple(:v)) { false }
+
+        rule(:string   => simple(:v)) { v.to_s }
+        rule(:number   => simple(:v)) { v.match(/[eE.]/) ? Float(v) : Integer(v) }
+        rule(:jsonpath => simple(:v)) do
+          results = JsonPath.on(input, v.to_s)
+          results.count < 2 ? results.first : results
```

Comment:
I think I want to change this to reuse Floe::Workflow::Path, but I might wait for a follow up PR.  This also doesn't handle the context at the moment.